src/MaIN.Services/Services/AgentService.cs

```diff
@@ -66,7 +67,7 @@ await notificationService.DispatchNotification(
     }
 
     public async Task<Agent> CreateAgent(Agent agent, bool flow = false, bool interactiveResponse = false,
-        InferenceParams? inferenceParams = null, MemoryParams? memoryParams = null)
+        InferenceParams? inferenceParams = null, MemoryParams? memoryParams = null, bool useCache = false)
```

Comment:
If the caching is for memory maybe we should default it to true? For sake of backwards compatibility

Reply:
Fair point, though caching make it more resource heavy, but users now have a choice to disable it at least. I will update thanks!

Reply:
done